tests/e2e/test_e2e.py

```diff
@@ -0,0 +1,116 @@
+from selenium import webdriver
+import os
+from threading import Thread
+import pytest
+import requests
+import trello_items as trello
+from dotenv import load_dotenv, find_dotenv
+import app
+import time
+from selenium.webdriver.support import expected_conditions as ec
+from selenium.webdriver.support.wait import WebDriverWait
+
+@pytest.fixture(scope='module')
+def test_app():
+    # Create the new board & update the board id environment variable 
+    # Use our test e2e config instead of the 'real' version 
+    file_path = find_dotenv('/.env')
+    load_dotenv(file_path, override=True)
+    board_id = create_trello_board('Test Board')
+    update_env_vars(board_id)
+    # construct the new application
+    application = app.create_app()
+    # start the app in its own thread.
+    thread = Thread(target=lambda: application.run(use_reloader=False)) 
+    thread.daemon = True
+    thread.start()
+    yield app
+    # Tear Down
+    thread.join(1) 
+    delete_trello_board(board_id)
+
+def update_env_vars(board_id):
+    os.environ['TRELLO_BOARD_ID'] = board_id
+    lists = trello.get_all_lists_on_board(board_id)
+    for list in lists:
+        if list['name'] == 'To Do':
+            os.environ['TODO_LIST_ID'] = list['id']
+        elif list['name'] == 'Doing':
+            os.environ['DOING_LIST_ID'] = list['id']
+        else:
+            os.environ['DONE_LIST_ID'] = list['id']
+
+
+@pytest.fixture(scope="module")
+def driver():
+    with webdriver.Firefox() as driver:
+        yield driver
+
+def test_task_journey(driver, test_app): 
+    driver.get('http://localhost:5000/')
+    assert driver.title == 'To-Do App'
+    #Create new item
+    els = driver.find_elements_by_tag_name("td")
+    assert len(els) == 0
+    driver.find_element_by_id("title-input").send_keys("Clean room")
+    driver.find_element_by_id("description-input").send_keys("Tidy room and wipe desk")
+    driver.find_element_by_id("create-btn").click()
+    time.sleep(2)
```

Comment:
Instead of all of these sleeps slowing down your test, you should be able to set driver.implicitly_wait(2) in the driver set up. That means it will wait **up to** 2 seconds when you look for an item, but will keep searching and stop waiting when it finds it (or raise an exception if the full time runs out).